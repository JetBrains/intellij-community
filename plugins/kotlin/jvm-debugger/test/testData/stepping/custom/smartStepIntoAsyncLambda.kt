package smartStepIntoAsyncLambda

fun lambdaInSequence() {
    val ranges = listOf(1, 2, 3)
    val seq = ranges.asSequence()
    // SMART_STEP_INTO_BY_INDEX: 1
    // STEP_OUT: 1
    // RESUME: 1
    //Breakpoint!
    seq
        .map {
            it
        }
        .forEach(::println)
}

fun lambdaInStream() {
    val ranges = listOf(1, 2, 3)
    // SMART_STEP_INTO_BY_INDEX: 1
    // STEP_OUT: 1
    //Breakpoint!
    ranges.stream()
        .map {
            it
        }
        .forEach(::println)
}

fun main() {
    lambdaInSequence()
    lambdaInStream()
}
