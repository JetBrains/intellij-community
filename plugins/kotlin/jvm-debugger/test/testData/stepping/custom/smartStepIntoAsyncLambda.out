LineBreakpoint created at smartStepIntoAsyncLambda.kt:10
LineBreakpoint created at smartStepIntoAsyncLambda.kt:22
Run Java
Connected to the target VM
smartStepIntoAsyncLambda.kt:10
smartStepIntoAsyncLambda.kt:12
smartStepIntoAsyncLambda.kt:14
smartStepIntoAsyncLambda.kt:22
smartStepIntoAsyncLambda.kt:24
smartStepIntoAsyncLambda.kt:27
Disconnected from the target VM

Process finished with exit code 0
